iceoryx2/src/config.rs

```diff
@@ -403,29 +489,40 @@ impl Config {
         Ok(ICEORYX2_CONFIG.get())
     }
 
-    /// Returns the global configuration. If the global configuration was not
-    /// [`Config::setup_global_config_from_file()`] it will load a default config. If
+    /// Returns the global configuration. If the global configuration was not yet loaded it will
+    /// load a default config by looking it up in the system. First it checks if a project local config file
+    /// exists, then if a config file in the user directory exist and then if a global config file exist. If
     /// [`Config::setup_global_config_from_file()`]
     /// is called after this function was called, no file will be loaded since the global default
     /// config was already populated.
     pub fn global_config() -> &'static Config {
+        let origin = "Config::global_config()";
         if !ICEORYX2_CONFIG.is_initialized() {
-            match Config::setup_global_config_from_file(unsafe {
-                &FilePath::new_unchecked(DEFAULT_CONFIG_FILE)
-            }) {
-                Ok(_) => (),
-                Err(ConfigCreationError::FailedToOpenConfigFile) => {
-                    debug!(from "Config::global_config()", "Default config file not found, populate config with default values.");
-                    ICEORYX2_CONFIG.set_value(Config::default());
-                }
-                Err(ConfigCreationError::FailedToReadConfigFileContents) => {
-                    warn!(from "Config::global_config()", "Default config file found but unable to read content, populate config with default values.");
-                    ICEORYX2_CONFIG.set_value(Config::default());
-                }
-                Err(ConfigCreationError::UnableToDeserializeContents) => {
-                    warn!(from "Config::global_config()", "Default config file found but unable to load data, populate config with default values.");
-                    ICEORYX2_CONFIG.set_value(Config::default());
+            let mut is_config_file_set = false;
+            if let Err(e) = Self::iterate_over_config_files(|config_file_path| {
+                match Config::setup_global_config_from_file(&config_file_path) {
+                    Ok(_) => {
+                        is_config_file_set = true;
+                        CallbackProgression::Stop
+                    }
+                    Err(ConfigCreationError::FailedToOpenConfigFile) => {
+                        CallbackProgression::Continue
```

Comment:
Does this error mean that the config file is not available or that the user does not have the access rights to open the config file?

Reply:
I introduced more fine grained error handling. If the file does not exist or the user does not have permission to access the file, the iteration will continue.

Reply:
No, if the user has not permission, it should not silently use another config file. This is error prone.